### ISSUE TITLE
ProtHint does not abort

### ISSUE THREAD

**Issue by KatharinaHoff** [gatech-genemark/ProtHint#2]
Hi Tomas!

ProtHint currently doesn't abort when things go wrong. Example log:

```
ProtHint Version 2.2.0
Copyright 2019, Georgia Institute of Technology, USA

Called from: /nas-hs/projs/braker/topaz/test_prothint/fly
Cmd: /home/katharina/git/ProtHint/bin/prothint.py --threads=10 /home/katharina/git/BRAKER/example/genome.fa proteins.fasta

[Fri Nov  8 13:13:30 2019] Running GeneMark-ES.
**/bin/sh: 1: /home/katharina/git/ProtHint/bin/../dependencies/GeneMarkES/bin/gmes_petap.pl: not found**
[Fri Nov  8 13:13:30 2019] GeneMark-ES finished.
[Fri Nov  8 13:13:30 2019] Translating GeneMark seeds to proteins
**error, file not found /home/katharina/git/ProtHint/bin/proteins_from_gtf.pl: /nas-hs/projs/braker/topaz/test_prothint/fly/GeneMark_ES/genemark.gtf**
[Fri Nov  8 13:13:31 2019] Translation of seeds finished
[Fri Nov  8 13:13:31 2019] Running DIAMOND
diamond v0.9.24.125 | by Benjamin Buchfink <buchfink@gmail.com>
Licensed under the GNU GPL <https://www.gnu.org/licenses/gpl.txt>
Check http://github.com/bbuchfink/diamond for updates.

#CPU threads: 12
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Database file: /nas-hs/projs/braker/topaz/test_prothint/fly/proteins.fasta
Opening the database file...  [0.006601s]
Loading sequences...  [3.37922s]
Masking sequences...  [30.9785s]
Writing sequences...  [0.556492s]
Hashing sequences...  [0.257572s]
Loading sequences...  [0.819388s]
Masking sequences...  [6.79235s]
Writing sequences...  [3.04667s]
Hashing sequences...  [0.058785s]
Loading sequences...  [1.9e-05s]
Writing trailer...  [0.018316s]
Closing the input file...  [8.8e-05s]
Closing the database file...  [2.36001s]
Database hash = 48fd6b4368d20beed208cb4adb93395d
Processed 2601995 sequences, 1211025059 letters.
Total time = 48.2742s
diamond v0.9.24.125 | by Benjamin Buchfink <buchfink@gmail.com>
Licensed under the GNU GPL <https://www.gnu.org/licenses/gpl.txt>
Check http://github.com/bbuchfink/diamond for updates.

#CPU threads: 10
Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)
Temporary directory: 
Opening the database...  [0.003002s]
#Target sequences to report alignments for: 250
Opening the input file... No such file or directory
 [0.020008s]
**Error: Error opening file ../gmes_proteins.faa**
[Fri Nov  8 13:14:19 2019] DIAMOND finished
[Fri Nov  8 13:14:19 2019] Preparing pairs for alignments
**error, file not found /home/katharina/git/ProtHint/bin/nucseq_for_selected_genes.pl: gene_stat.yaml**
[Fri Nov  8 13:14:19 2019] Preparation of pairs finished
**error, file not found /home/katharina/git/ProtHint/bin/run_spliced_alignment.pl: ../nuc.fasta**
[Fri Nov  8 13:14:19 2019] Processing the output
grep: Spaln/spaln.gff: No such file or directory
grep: Spaln/spaln.gff: No such file or directory
grep: Spaln/spaln.gff: No such file or directory
grep: Spaln/spaln.gff: No such file or directory
[Fri Nov  8 13:14:20 2019] Output processed
[Fri Nov  8 13:14:20 2019] ProtHint finished.
```
To me, it would have made sense if the process died with an non-zero exit code after the first error, already.

Katharina

**Comment by tomasbruna** [comment]
Hi Katharina,

this definitely makes sense, I will add a wrapper around all system calls to terminate the program if things go wrong. Thanks for the suggestion.

Tomas

**Comment by tomasbruna** [comment]
Implemented in 202d4b7b4105810e1910d4d094c23c1ab59c9b48 and https://github.com/gatech-genemark/ProtHint/commit/2d4b8d47ef7f362d9a50db5b396168e0d38fef95
